### PR TITLE
NSLS-II Improvements to banner, users

### PR DESCRIPTION
**Pull request by stuwilkins**
Support for banner file.

Good: Reads from a file and file is settable from env variables
Bad: Banner text is read each time prompt is drawn (each entry of password) 

Would need to do either global variable for banner text or do more surgery, any ideas?

### CODE REVIEW

**Comment by stuwilkins** [comment]
Ok ... embrace global variables .........

**Comment by stuwilkins** [comment]
@stuartcampbell @cryos This is now ready for review.

**Comment by stuwilkins** [comment]
P.S. Lets not squash merge this time....

File: README-NSLS2.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+[![Build Status](https://dev.azure.com/nsls-ii/xsecurelock/_apis/build/status/NSLS-II.xsecurelock?branchName=refs%2Fpull%2F3%2Fmerge)](https://dev.azure.com/nsls-ii/xsecurelock/_build/latest?definitionId=5&branchName=refs%2Fpull%2F3%2Fmerge)
+
+ This outlines the changes made to this version for NSLS-II specific changes.
+ 
+ All changes to the upstream code are contained within compiler
+ pre-processing directives set through _autoconf_. This enables the upstream
+ and changed versions to both be built through the same code base. It also
+ allows for better auditing of changes.
+
+In addition to these changes, CI is provided an Azure Pipelines and upstream
```
Comment (cryos) [inline README-NSLS2.md]: I think you mean "provided on Azure Pipelines" here.

File: banner.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        if (*num_lines >= max_lines) {
+            rtn = 2;
+            break;
+        }
+        line++;
+    }
+
+    fclose(fp);
+    return rtn;
+}
```
Comment (cryos) [inline banner.c]: Lots of no newline at end of file, but I think most compilers are over this by now.

**Review by cryos — APPROVED** [review]
A couple of very minor points but it looks good to me. I don't know these APIs that well, and don't drop to C that often so you may prefer another set of eyes on this if you want a deeper review.

File: banner.c (hunk 2 above)
Comment (stuwilkins) [reply banner.c]: Thanks.... I hadn't noticed that, but it was a vscode thing that it was stripping them out! I think this is now fixed?

File: README-NSLS2.md (hunk 1 above)
Comment (stuwilkins) [reply README-NSLS2.md]: Yes! Fixed

File: banner.c (hunk 2 above)
Comment (cryos) [reply banner.c]: Yes - it looks like they are all fixed in the current diff of the branch.

**Review by stuartcampbell — APPROVED** [review]
I can't see any major issues.  Looks good.

**Pull request merged**